pkg/resource/pod_distruption_budget.go

```diff
@@ -0,0 +1,94 @@
+package resource
+
+import (
+	"errors"
+	"fmt"
+
+	policy "k8s.io/api/policy/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// PdbBuilder models the PodDistruptionBudget that the
+// operator maintains.
+type PdbBuilder struct {
+	*Cluster
+
+	Selector map[string]string
+}
+
+// Build creates a policy.PodDisruptionBudget for the
+// StatefulSet.
+func (b PdbBuilder) Build(obj runtime.Object) error {
+	pdb, ok := obj.(*policy.PodDisruptionBudget)
+	if !ok {
+		return errors.New("failed to cast to PDB object")
+	}
+
+	// if we only have one Node we cannot have a PDB
+	// TODO we need to validate this in the CRD API
+	if b.Spec().Nodes == 1 {
+		return nil
```

Comment:
This will never be called as the actor always checks for `> 1`

Reply:
Removed